mitmproxy/addons/next_layer.py

```diff
@@ -220,10 +220,18 @@ def _ignore_connection(
             client_hello := self._get_client_hello(context, data_client)
         ) and client_hello.sni:
             hostnames.append(client_hello.sni)
-
+        # If the client data is not a TLS record, try to extract the domain from the HTTP request
+        else:
+            if isinstance(data_client, bytes):
+                host = self._extract_host(data_client)
+                if host:
+                    hostnames.append(host)
         if not hostnames:
             return False
 
+        # Special handling for wireguard mode: if the hostname is "10.0.0.53", do not ignore the connection
+        if "wireguard" in ctx.options.mode and "10.0.0.53" in hostnames:
```

Comment:
We can do a slightly more precise check here, like this:
https://github.com/mitmproxy/mitmproxy/blob/439e5264eb8aaff74aeb80588b86e13b9f0c320e/mitmproxy/addons/dns_resolver.py#L147-L148

flow.client_conn -> context.client and the same for server. :)

mitmproxy can spawn multiple proxy servers now, and this makes sure we only target the correct one.
We can also move this to the top of the method as it does not depend on collecting `hostnames`.

Reply:
Hello @mhils,

Thank you for your valuable feedback. I greatly appreciate your suggestions, they make perfect sense.